### ISSUE TITLE
Charsets and metadata on input.harbor

### ISSUE THREAD

**Issue by Calypso971** [savonet/liquidsoap#3482]
Not sure if I'm looking at a user-error or a bug here. 

In pre 2.2 liquidsoap (Debian/Bookworm) sending metadata with a stream (icecast protocol) was working fine:

```
2023/10/19 20:45:40 [harbor:4] New client on port xxxx: aa.bb.cc.dd
2023/10/19 20:45:40 [harbor:4] Method: GET, uri: /admin/metadata?mount=/nonstop&mode=updinfo&song=David%20McWilliams%20-%20Can%20I%20Get%20There%20By%20Candlelight&charset=utf-8, protocol: HTTP/1.1
2023/10/19 20:45:40 [harbor:4] Header: Host, value: ee.ff.gg.hh:xxxx.
2023/10/19 20:45:40 [harbor:4] Header: Connection, value: Keep-Alive.
2023/10/19 20:45:40 [harbor:4] Header: Accept-Encoding, value: gzip, deflate.
2023/10/19 20:45:40 [harbor:4] Header: Accept-Language, value: en-NL,*.
2023/10/19 20:45:40 [harbor:4] Header: User-Agent, value: Mozilla/5.0.
2023/10/19 20:45:40 [harbor:4] HTTP GET request on /admin/metadata.
2023/10/19 20:45:40 [harbor:4] HTTP Arg: mount, value: /mymount.
2023/10/19 20:45:40 [harbor:4] HTTP Arg: charset, value: utf-8.
2023/10/19 20:45:40 [harbor:4] HTTP Arg: song, value: David McWilliams - Can I Get There By Candlelight.
2023/10/19 20:45:40 [harbor:4] HTTP Arg: mode, value: updinfo.
2023/10/19 20:45:40 [harbor:4] Request to update metadata for mount /nonstop on port 8920
2023/10/19 20:45:40 [harbor:4] Client logged in.
2023/10/19 20:45:40 [mymount:3] New metadata chunk ? -- David McWilliams - Can I Get There By Candlelight.
```

... and there's my metadata

In 2.2 of course things concerning characters ets have changed and I don't get the metadata anymore:

```
2023/10/19 20:40:22 [harbor:4] New client on port xxxx: aa.bb.cc.dd
2023/10/19 20:40:22 [harbor:4] Method: GET, uri: /admin/metadata?mount=/nonstop&mode=updinfo&song=Pet%20Shop%20Boys%20-%20It's%20A%20Sin&charset=utf-8, protocol: HTTP/1.1
2023/10/19 20:40:22 [harbor:4] Header: Host, value: ee.ff.gg.hh:xxxx.
2023/10/19 20:40:22 [harbor:4] Header: Connection, value: Keep-Alive.
2023/10/19 20:40:22 [harbor:4] Header: Accept-Encoding, value: gzip, deflate.
2023/10/19 20:40:22 [harbor:4] Header: Accept-Language, value: en-NL,*.
2023/10/19 20:40:22 [harbor:4] Header: User-Agent, value: Mozilla/5.0.
2023/10/19 20:40:22 [harbor:4] http/unix GET request on /admin/metadata.
2023/10/19 20:40:22 [harbor:4] http/unix Arg: mount, value: /mymount.
2023/10/19 20:40:22 [harbor:4] http/unix Arg: charset, value: utf-8.
2023/10/19 20:40:22 [harbor:4] http/unix Arg: song, value: Pet Shop Boys - It's A Sin.
2023/10/19 20:40:22 [harbor:4] http/unix Arg: mode, value: updinfo.
2023/10/19 20:40:22 [harbor:4] Request to update metadata for mount /nonstop on port 8920
2023/10/19 20:40:22 [harbor:4] Client logged in.
2023/10/19 20:40:22 [harbor:4] Charset_base.Unknown_encoding("utf-8")
```

and that's it. The metadata is sent, it's parsed, however, I got an "Unknown_encoding" and it's seems not to be processed as metadata in Liquidsoap.

I have been fiddling around with the metadata settings on my harbor and assured that on Debian UTF-8 is used, but upto now I haven't been able to get things to work correctly.

Before changing stuff (i.e. playing around with things like metadata_charset= etc), my input harbor was simply:

```
stream = input.harbor("mymount", id="mymount", port=xxxx,user="source",password="mysupersecretpassword", buffer=5.,max=20.)
```

and after that is was (in my testscript) just sent out to an `output.icecast`. 

Any idea what I'm missing? Because I think it's most likely a very simple thing...

[Tested versions: precompiled 2.1.4 and 2.2.1]

**Comment by toots** [comment]
Hi! This is a curious issue. How do you install liquidsoap? Is that a windows build?

**Comment by Calypso971** [comment]
No, not mentioned directly, but I'm using Debian/Bookworm and the rpecompiled builds for 2.1.4 and 2.2.1:

9006488 Mar  3  2023 liquidsoap_2.1.4-debian-bookworm-1_amd64.deb
7938276 Sep  5 12:12 liquidsoap_2.2.1-debian-bookworm-1_amd64.deb

**Comment by toots** [comment]
Thanks for submitting. This should be fixed in `rolling-release-v2.2.x` and `main`
